### PR TITLE
Remove generic type parameter in LS-package inheritance hierarchy.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
This parameter was used to ensure type-safety with subclasses.  However, no one was
benefitting from knowing the strong type of the subclass (in C#/VB), so this ended up being
non-helpful.  I've removed it to help simplify a lot of the code we have around
packages/language-services/projects.

Fixes https://github.com/dotnet/roslyn/issues/14270

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
tagging @mavasani @jasonmalinowski 

@paulvanbrenk @vladima @minestarks Will this affect you?  Your code must be passing along "TypeScriptProject" to this generic parameter today.  However, does your code ever depend on getting projects back of exactly this type?  If so, are you opposed to just manually casting from AbstractProject to TypeScriptProject in that case?

Also tagging @mgoertz-msft and @OmarTawfik   If you derive from AbstractPackage/AbstractLanguageService this will affect you in the same way.  Let me know if you depended on the type-safety here and if it critical to preserve it.  Otherwise, this helps simplify things a bunch and gets rid of a bit of hte generic-soup we have now.

**Comment by OmarTawfik** [comment]
@CyrusNajmabadi changed F# language service to use the `AbstractLanguageService<TPackage, TLanguageService>`, and import a `IWorkspaceProjectContextFactory`.

**Comment by CyrusNajmabadi** [comment]
test windows_debug_unit64_prtest  please

**Comment by CyrusNajmabadi** [comment]
test windows_release_unit64_prtest  please

**Comment by CyrusNajmabadi** [comment]
@OmarTawfik I'm not sure i understand your comment.  Are you saying you're ok with my change, or you're *not* ok wiht my change?

Thanks!

**Comment by OmarTawfik** [comment]
@CyrusNajmabadi I'm saying we dropped using the three-type version of these classes, and are using the two-type version now. Check [this](https://github.com/OmarTawfik/visualfsharp/commit/0591eb196dc8728b78886a525090768f4f9c5aae) commit (still WIP).

**Comment by CyrusNajmabadi** [comment]
Omar and i talked offline.  Prior to this PR is it not supported to just derive from the 2-type-parameter versions.  With this PR it will be supported (as teh 3-type-parameter version is just going away).

**Comment by paulvanbrenk** [comment]
LGTM

File: src/VisualStudio/Core/Def/Implementation/AbstractVsTextViewFilter`3.cs [hunk 1]
```diff
@@ -16,15 +16,14 @@
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation
 {
-    internal abstract class AbstractVsTextViewFilter<TPackage, TLanguageService, TProject> : AbstractVsTextViewFilter, IVsTextViewFilter, IVsReadOnlyViewNotification
-        where TPackage : AbstractPackage<TPackage, TLanguageService, TProject>
-        where TLanguageService : AbstractLanguageService<TPackage, TLanguageService, TProject>
-        where TProject : AbstractProject
+    internal abstract class AbstractVsTextViewFilter<TPackage, TLanguageService> : AbstractVsTextViewFilter, IVsTextViewFilter, IVsReadOnlyViewNotification
```
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Def/Implementation/AbstractVsTextViewFilter`3.cs]: This file is still called `3 implying it's the three-argument form.

File: src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractLanguageService`3.IVsContainedLanguageFactory.cs [hunk 2]
```diff
@@ -3,15 +3,16 @@
...
 using System.Runtime.InteropServices;
+using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem;
 using Microsoft.VisualStudio.Shell.Interop;
 using Microsoft.VisualStudio.TextManager.Interop;
 using VsServiceProvider = Microsoft.VisualStudio.OLE.Interop.IServiceProvider;
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.LanguageService
 {
-    internal abstract partial class AbstractLanguageService<TPackage, TLanguageService, TProject> : IVsContainedLanguageFactory
+    internal abstract partial class AbstractLanguageService<TPackage, TLanguageService> : IVsContainedLanguageFactory
```
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractLanguageService`3.IVsContainedLanguageFactory.cs]: FileName is still AbstractLanguageService`3....

File: src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractLanguageService`3.IVsDebuggerEvents.cs [hunk 3]
```diff
@@ -1,13 +1,12 @@
...
 
 using System;
-using Microsoft.VisualStudio;
 using Microsoft.VisualStudio.LanguageServices.Implementation.Debugging;
 using Microsoft.VisualStudio.Shell.Interop;
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.LanguageService
 {
-    internal abstract partial class AbstractLanguageService<TPackage, TLanguageService, TProject> : IVsDebuggerEvents
+    internal abstract partial class AbstractLanguageService<TPackage, TLanguageService> : IVsDebuggerEvents
```
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractLanguageService`3.IVsDebuggerEvents.cs]: File name is still `3...

File: src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractLanguageService`3.IVsImmediateStatementCompletion2.cs [hunk 4]
```diff
@@ -13,14 +13,14 @@
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.LanguageService
 {
-    internal abstract partial class AbstractLanguageService<TPackage, TLanguageService, TProject> : IVsImmediateStatementCompletion2
+    internal abstract partial class AbstractLanguageService<TPackage, TLanguageService> : IVsImmediateStatementCompletion2
```
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractLanguageService`3.IVsImmediateStatementCompletion2.cs]: File name is still `3...

File: src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractLanguageService`3.IVsLanguageDebugInfo.cs [hunk 5]
```diff
@@ -8,7 +8,7 @@
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.LanguageService
 {
-    internal partial class AbstractLanguageService<TPackage, TLanguageService, TProject> : IVsLanguageDebugInfo
+    internal partial class AbstractLanguageService<TPackage, TLanguageService> : IVsLanguageDebugInfo
```
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractLanguageService`3.IVsLanguageDebugInfo.cs]: File name is still `3...

**Comment by CyrusNajmabadi** [comment]
Ok.  I'm going to merge now.  FYI @vladima @paulvanbrenk this is not going to be part of the current RI that we're trying to get it.  So this will not be inserted today.  So please don't try to insert matching bits until it goes in :)

**Pull request merged**